.github/workflows/ci.yml

```diff
@@ -34,52 +34,70 @@ jobs:
           pip install wheel
           curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
 
+      - name: Setup cache variables
+        shell: bash
+        id: cache-vars
+        run: |
+          export PATH="$HOME/.poetry/bin:$PATH"
+          echo "::set-output name=poetry-dir::$(poetry config cache-dir)"
+          echo "::set-output name=date::$(/bin/date -u "+%m%w%Y")"
+
+      - name: Setup Poetry cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.cache-vars.outputs.poetry-dir }}
+          key: ${{ runner.os }}-poetry-${{ hashFiles('poetry.lock') }}-${{ matrix.python }}
+
+      - name: Install and cache ffmpeg (all OS)
+        uses: FedericoCarboni/setup-ffmpeg@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+        id: setup-ffmpeg
+
       - name: Install system dependencies (Linux)
         if: runner.os == 'Linux'
         run: |
           sudo apt update
-          sudo apt install -y ffmpeg
           sudo apt-get -y install texlive texlive-latex-extra texlive-fonts-extra texlive-latex-recommended texlive-science texlive-fonts-extra tipa python-opengl libpango1.0-dev xvfb
           echo "$HOME/.poetry/bin" >> $GITHUB_PATH
           # start xvfb in the background
           sudo /usr/bin/Xvfb $DISPLAY -screen 0 1280x1024x24 &
 
+      - name: Setup Brew cache (MacOS)
+        id: cache-brew
+        if: runner.os == 'macOS'
+        uses: actions/cache@v2
+        with:
+          path: ~/brew-cache
+          key: test8-brew-${{ matrix.python }}-${{ steps.cache-vars.outputs.date }}
+
       - name: Install system dependencies (MacOS)
         if: runner.os == 'macOS'
         run: |
-          brew update && brew upgrade basictex && brew cleanup
-          brew install openssl readline ffmpeg pyenv pyenv-virtualenv
+          brew update -q && brew upgrade basictex && brew cleanup
+          brew install openssl readline pyenv pyenv-virtualenv
           brew install --cask basictex
           eval "$(/usr/libexec/path_helper -s)"
           sudo tlmgr update --self
-          brew install pkg-config libffi pango glib
           sudo tlmgr install standalone preview doublestroke relsize fundus-calligra wasysym physics dvisvgm rsfs wasy cm-super
-          echo "/Library/TeX/texbin" >> $GITHUB_PATH
-          echo "$HOME/.poetry/bin" >> $GITHUB_PATH
+          brew install pkg-config libffi glib pango
```

Comment:
Maybe just Cairo? Why do you need Pango?
```suggestion
          brew install pkg-config cairo
```

Reply:
I think pango is for text and font rendering.

Reply:
It appears that ffmpeg and cairo share some dependencies, so installing ffmpeg from binaries and caching it by using the action still requires cairo to install those dependencies separately.
![image](https://user-images.githubusercontent.com/24194862/113936878-6ba18b80-97c6-11eb-9df6-352cf5585cf5.png)
![image](https://user-images.githubusercontent.com/24194862/113936886-6f351280-97c6-11eb-9866-25d3424f97ca.png)